src/clj/commiteth/scheduler.clj

```diff
@@ -248,6 +276,43 @@
     (neg? n) (- n)
     :else n))
 
+(defn contract-confirmation-logs [contract-address]
+  "retrives all log events for the confirmation topic since contract creation"
+  (some-> contract-address
+      issues/get-issue-by-contract-address
+      :transaction_hash
+      eth/get-transaction-by-hash
+      :blockNumber
+      (eth/get-logs contract-address [(:confirmation multisig/topics)])))
+
+(defn hash-in-logs?
+  "return true if the transaction hash is present in the queryable blockchain"
+  [hash logs]
+  (some #(= hash (:transactionHash %)) logs))
+
+(defn execution-status [execute-hash contract-address]
+  "check to see if a given execute-hash has been confirmed"
+  (log/infof "checking contract for logs containing %s" execute-hash)
+  (let [logs (contract-confirmation-logs contract-address)]
+    (hash-in-logs? execute-hash logs)))
+
+
+(defn poll-transaction-logs [execute-hash contract-address]
+  "check for execution hash in logs for a few minutes"
+  (let [found? (promise)
+        intervals (take 6
+                    (periodic-seq (t/now)
+                                  (t/seconds 30)))]
+    ;; polling will be slow but if we want to move to an event driven
+    ;; model then we can listen for events, rather than logs, once we're
+    ;; using a geth node again
+    (chime-at intervals
+              (fn [time]
+                (when (execution-status execute-hash contract-address)
+                  (deliver found? true)))
+              {:on-finished (fn []
+                              (deliver found? false))})
+    @found?))
```

Comment:
Good point! As with a few of the other changes that were introduced this will be addressed by removing this functionality entirely. The big shift that actually vitaliy pointed out is that reframe is already reloading data via a series of api calls every minute, so revocation can take advantage of that rather than having to do its own polling.